community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/pipes/QueryState.scala

```diff
@@ -56,8 +57,10 @@ class QueryState(val query: QueryContext,
 
   def readTimeStamp(): Long = timeReader.getTime
 
-  def  getParam(key: String): AnyValue =
-    params.getOrElse(key, throw new ParameterNotFoundException("Expected a parameter named " + key))
+  def  getParam(key: String): AnyValue = {
+    if (!params.containsKey(key)) throw new ParameterNotFoundException("Expected a parameter named " + key)
+    params.get(key)
```

Comment:
Not sure about this one? Will give 2 map lookups instead of one, and more code. Please motivate =)

Reply:
It did two lookups before as well scala just hid it for us. I agree that the code is a little bit ugglier but we cannot do

Reply:
Ok. Nasty scala. Let's keep it for now then.